packages/common/src/util.js

```diff
@@ -49,3 +68,99 @@ export function normalizeOauthConfig(configuration) {
 
   return configuration;
 }
+
+function separateUrl(fullUrl) {
+  const urlObject = new URL(fullUrl);
+  const baseUrl = `${urlObject.protocol}//${urlObject.host}`;
+  const path = urlObject.pathname;
+  return { baseUrl, path };
+}
+
+export async function request(method, fullUrl, options = {}) {
+  const { baseUrl, path } = separateUrl(fullUrl);
+  const defaultOptions = {
+    headers: {},
+    query: undefined,
+    timeout: {},
+    tls: {},
+    body: undefined,
+    errors: {
+      401: 'Unauthorised',
+      404: 'Not found',
+    },
+  };
+
+  const { headers, query, body, errors, timeout, tls } = {
+    ...defaultOptions,
+    ...options,
+  };
+
+  if (!headers['Content-Type'] && body) {
+    headers['Content-Type'] = 'application/json';
+  }
+
+  const client = getClient(baseUrl);
+
+  const response = await client.request({
+    path: path,
+    query: query,
+    method: method,
+    headers: headers,
+    errors: errors,
+    body: body ? JSON.stringify(body) : undefined,
+    throwOnError: false,
+  });
+
+  const responseBody = await readResponseBody(response);
+
+  if (response.statusCode >= 400) {
```

Comment:
If I wanted to do: `errors: { 204: "content not found" }`, it wouldn't throw. 

You should write a unit test to prove this, then fix it.